File: BackEnd/src/main/java/tn/encar/gestnotes/entities/Etudiant.java
package tn.encar.gestnotes.entities;


import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
public class Etudiant extends Personne{

	private int numInscri;
	private int niveau;
	private TypeFormation typeFormation ;

}

File: FrontEnd/src/app/models/Etudiant.ts
import { Absence } from "./Absence";
import { Classe } from "./Classe";
import { Note } from "./Note";
import { Personne } from "./Personne";
import { TypeFormation } from "./TypeFormation";

export class Etudiant extends Personne{

    numInscri: number;
    classe: Classe;
    typeFormation: TypeFormation;
    absences: Absence[];
    notes: Note[];

}
